Updating Pull Request template checklist (#1230)
haddocking/haddock3: .github/pull_request_template.md

- [ ] Tests added for the new code
- [ ] Documentation added for the code changes
- [ ] Modifications / enhancements are reflected on the [haddock3 user-manual](https://github.com/haddocking/haddock3-user-manual)

> - [ ] [CHANGELOG.md](CHANGELOG.md) is updated to incorporate new changes

Author: ```suggestion
- [ ] `CHANGELOG.md` is updated to incorporate new changes
```

## Related Issue
<!-- If this PR is related to an issue, please link it here -->

> <!-- If this PR is has implication on the haddock3 user manual, please link the PR here -->

Reviewer: ```suggestion
<!-- If this PR is related to the haddock3 user manual, please link the PR here -->
```
